src/Caching/Impl/ModuleDatabase.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using LiteDB;
+using Microsoft.Python.Analysis.Analyzer;
+using Microsoft.Python.Analysis.Caching.Models;
+using Microsoft.Python.Analysis.Dependencies;
+using Microsoft.Python.Analysis.Modules;
+using Microsoft.Python.Analysis.Types;
+using Microsoft.Python.Core;
+using Microsoft.Python.Core.IO;
+using Microsoft.Python.Core.Logging;
+using Microsoft.Python.Parsing.Ast;
+
+namespace Microsoft.Python.Analysis.Caching {
+    internal sealed class ModuleDatabase : IModuleDatabaseService {
+        private const int _databaseFormatVersion = 1;
```

Comment:
Would this be better named `DatabaseFormatVersion` to indicate that this isn't a member of the database (but some global-ish constant)?